From ed4e8b5d4ad6ffcddb755571303c077fff227ea3 Mon Sep 17 00:00:00 2001
From: Quirin Gylstorff <quirin.gylstorff@siemens.com>
Date: Fri, 16 Jun 2023 19:43:38 +0200
Subject: [PATCH 3/4] d/patches: Add patch to add the build version to swupdate

The build version is only set if build in a git repository
this patch allows to use the version information stored in
the Makefile.

Signed-off-by: Quirin Gylstorff <quirin.gylstorff@siemens.com>
---
 ...Set-version-if-git-returns-no-informa.diff | 46 +++++++++++++++++++
 debian/patches/series                         |  1 +
 debian/rules                                  |  2 +
 3 files changed, 49 insertions(+)
 create mode 100644 debian/patches/Makefile.flags-Set-version-if-git-returns-no-informa.diff

diff --git a/debian/patches/Makefile.flags-Set-version-if-git-returns-no-informa.diff b/debian/patches/Makefile.flags-Set-version-if-git-returns-no-informa.diff
new file mode 100644
index 00000000..7f964fb7
--- /dev/null
+++ b/debian/patches/Makefile.flags-Set-version-if-git-returns-no-informa.diff
@@ -0,0 +1,46 @@
+From bc50744125e8686ec2eb7e5676522da98e51601f Mon Sep 17 00:00:00 2001
+From: Quirin Gylstorff <quirin.gylstorff@siemens.com>
+Date: Sat, 17 Jun 2023 17:56:18 +0200
+Subject: [PATCH] Makefile.flags: Set version if git returns no information
+
+This set the swupdate version if build in a non git environment.
+E.g. pristine-tar build or from a release tarball.
+
+Also allows to set the EXTRAVERSION from the command line.
+
+Signed-off-by: Quirin Gylstorff <quirin.gylstorff@siemens.com>
+---
+ Makefile       | 2 +-
+ Makefile.flags | 3 +++
+ 2 files changed, 4 insertions(+), 1 deletion(-)
+
+diff --git a/Makefile b/Makefile
+index 09d03a34..1004945c 100644
+--- a/Makefile
++++ b/Makefile
+@@ -5,7 +5,7 @@
+ VERSION = 2023
+ PATCHLEVEL = 05
+ SUBLEVEL = 0
+-EXTRAVERSION =
++EXTRAVERSION ?=
+ NAME =
+ 
+ IPCLIB_VERSION = 0.1
+diff --git a/Makefile.flags b/Makefile.flags
+index 2a021c89..8eada8fc 100644
+--- a/Makefile.flags
++++ b/Makefile.flags
+@@ -7,6 +7,9 @@
+ # ==========================================================================
+ SWU_DIR = $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+ SWU_VER = $(shell git -C $(SWU_DIR) describe --tags --always --dirty)
++ifeq (,$(SWU_VER))
++	SWU_VER = $(VERSION).$(PATCHLEVEL).$(SUBLEVEL)$(EXTRAVERSION)
++endif
+ export SWU_VER
+ SKIP_STRIP ?= n
+ 
+-- 
+2.39.2
+
diff --git a/debian/patches/series b/debian/patches/series
index 5638bbf5..12324569 100644
--- a/debian/patches/series
+++ b/debian/patches/series
@@ -1,3 +1,4 @@
 Link-config-to-swupdate-www-path.diff
 Replace-Font-Awesome-5-with-Fork-Awesome.diff
 use-gcc-compiler.diff
+Makefile.flags-Set-version-if-git-returns-no-informa.diff
diff --git a/debian/rules b/debian/rules
index 8cac6359..566a21bd 100755
--- a/debian/rules
+++ b/debian/rules
@@ -17,6 +17,8 @@ export DH_WITH=--with lua
 export HAVE_LUA=y
 endif
 
+DEB_CHANGELOG_VERSION := $(shell dpkg-parsechangelog | sed -n -e 's/^Version: //p' | cut -d + -f 2)
+export EXTRAVERSION=($(DEB_CHANGELOG_VERSION))
 BU_SWU=
 ifeq (linux,$(DEB_HOST_ARCH_OS))
 	BU_SWU += -VBuilt-Using="$(shell dpkg-query -f '$${source:Package} (= $${source:Version}), ' -W libmtd-dev libubi-dev)"
-- 
2.39.2

